src/components/Tokens/TokenDetails/BalanceSummary.tsx

```diff
@@ -61,49 +69,214 @@ const StyledNetworkLabel = styled.div`
-  const { account, chainId } = useWeb3React()
-  const theme = useTheme()
-  const { label, color } = getChainInfo(asSupportedChain(chainId) ?? ChainId.MAINNET)
-  const balance = useCurrencyBalance(account, token)
-  const { formatCurrencyAmount } = useFormatter()
+interface BalanceProps {
+  token?: Currency
+  chainId: ChainId
+  balance?: CurrencyAmount<Currency> // TODO(WEB-3026): only used for pre-Info-project calculations, should remove after project goes live
+  gqlBalance?: TokenBalance
+  tokenSymbol?: string
+  color?: string
+  chainName?: string
+  onClick?: () => void
+  isInfoTDPEnabled?: boolean
+}
+const Balance = (props: BalanceProps) => {
+  const { token, chainId, balance, gqlBalance, tokenSymbol, color, chainName, onClick, isInfoTDPEnabled } = props
+  const { formatCurrencyAmount, formatNumber } = useFormatter()
+  const currencies = useMemo(() => [token], [token])
+
   const formattedBalance = formatCurrencyAmount({
     amount: balance,
     type: NumberType.TokenNonTx,
   })
   const formattedUsdValue = formatCurrencyAmount({
     amount: useStablecoinValue(balance),
-    type: NumberType.FiatTokenStats,
+    type: NumberType.PortfolioBalance,
+  })
+  const formattedGqlBalance = formatNumber({
+    input: gqlBalance?.quantity,
+    type: NumberType.TokenNonTx,
+  })
+  const formattedUsdGqlValue = formatNumber({
+    input: gqlBalance?.denominatedValue?.value,
+    type: NumberType.PortfolioBalance,
   })
 
-  const currencies = useMemo(() => [token], [token])
+  if (isInfoTDPEnabled) {
+    return (
+      <BalanceRow onClick={onClick}>
+        <PortfolioLogo currencies={currencies} chainId={chainId} size="2rem" />
+        <BalanceAmountsContainer isInfoTDPEnabled>
+          <BalanceItem>
+            <ThemedText.BodyPrimary>{formattedUsdGqlValue}</ThemedText.BodyPrimary>
+          </BalanceItem>
+          <BalanceItem>
+            <ThemedText.BodySecondary>{formattedGqlBalance}</ThemedText.BodySecondary>
+          </BalanceItem>
+        </BalanceAmountsContainer>
+      </BalanceRow>
+    )
+  } else {
+    return (
+      <BalanceRow>
+        <PortfolioLogo currencies={currencies} chainId={chainId} size="2rem" />
+        <BalanceContainer>
+          <BalanceAmountsContainer>
+            <BalanceItem>
+              <ThemedText.SubHeader>
+                {formattedBalance} {tokenSymbol}
+              </ThemedText.SubHeader>
+            </BalanceItem>
+            <BalanceItem>
+              <ThemedText.BodyPrimary>{formattedUsdValue}</ThemedText.BodyPrimary>
+            </BalanceItem>
+          </BalanceAmountsContainer>
+          <StyledNetworkLabel color={color}>{chainName}</StyledNetworkLabel>
+        </BalanceContainer>
+      </BalanceRow>
+    )
+  }
+}
+
+const ConnectedChainBalanceSummary = ({
+  connectedChainBalance,
+  chainId,
+  token,
+  theme,
+}: {
+  connectedChainBalance?: CurrencyAmount<Currency>
+  chainId?: ChainId
```

Comment:
you should be able to derive token and chainName/chainId from `pathname`, you can see how we do it in `src/pages/TokenDetails/index.tsx`
```
const { tokenAddress, chainName } = useParams<{
    tokenAddress: string
    chainName?: string
  }>()
  const chain = validateUrlChainParam(chainName)